Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsRequestValueGetterService.cs

```diff
@@ -10,16 +10,15 @@ public class BotCommandsRequestValueGetterService
     {
         public object GetValueByName(string key, bool isList, DiscordRequest request, BotCommandTemplate template)
         {
-            var result = request.Arguments.FirstOrDefault(a => a.Name.ToLowerInvariant() == key.ToLowerInvariant());
-            if (result == null)
+            var argType = template.Properties.First(x => x.Name.ToLowerInvariant() == key.ToLowerInvariant()).Type;
+            var result = request.Arguments.FirstOrDefault(a => a.Name?.ToLowerInvariant() == key.ToLowerInvariant());
+            if (argType == BotCommandPropertyType.Bool)
             {
-                return null;
+                return result == null ? bool.FalseString : bool.TrueString;
```

Comment:
czym są FalseString i TrueString?

Reply:
Wiesz zawsze można spojrzyć na MSDN ;) 
Ale chodzi o to, że jeśli nie podano argumentu, który jest bool to odpowiednie property będzie miało false (bo FalseString będzie przekonwertowane na false), a jeśli taki argument został podany to property będzie miało true. 
Wcześniej było, że TrueString dla podanego argument a gdy nie podano to zwracano null co powodowało, że i tak wyszło false w owym property, ale trochę się pozmieniało w parsowaniu przez dodanie obsługi domyślnych property innych niż Bool i niestety musiałem to tak zrobić.

Reply:
a czemu nie wystarczy zwracanie zwykłego true/false? 
`return result != null`

Reply:
wydaje mi się, że obecnie jest prościej, ale nie wiem, bo gdyby było true/false to byłoby trzeba dalej w parsowaniu pozmieniać.